src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -252,8 +252,9 @@ internal string SolutionFileDirectory
 
         #region Methods
 
-        internal bool ProjectShouldBuild(string projectFile)
+        public bool ProjectShouldBuild(string projectFile)
```

Comment:
@mrward is this something we might be able to use in vsmac as well?

Reply:
VS Mac has its own logic here - the project filtered out by the slnf file is not loaded and so would not be restored.

Reply:
We ultimately decided this isn't the way to go, so it will remain risky to rely on ProjectShouldBuild staying as it is.